unlock-app/src/components/interface/locks/Manage/index.tsx

```diff
@@ -230,14 +230,27 @@ const TopActionBar = ({ lockAddress, network }: TopActionBarProps) => {
 }
 
 export const ManageLockPage = () => {
-  const { network: walletNetwork } = useAuth()
+  const { network: walletNetwork, changeNetwork } = useAuth()
   const { query } = useRouter()
 
   const { address, network } = query ?? {}
 
   const lockNetwork = parseInt(network as string)
   const lockAddress = address as string
 
+  // let's force to switch network based on the lockAddress
+  const switchToCurrentNetwork = async () => {
+    const differentNetwork = walletNetwork != network
+
+    if (differentNetwork) {
+      await changeNetwork(parseInt(`${network}`))
+    }
+  }
+
+  useEffect(() => {
+    switchToCurrentNetwork()
```

Comment:
I wonder if we shoud block rendering in that case too just to avoid issues if someone closes the modal?

Reply:
I think it is not necessary. We will replace it later.

Reply:
ok LG